From 1f8a20c8309bb0e4eeeeb4e43f9688ba847502e1 Mon Sep 17 00:00:00 2001
From: Konstantin Porotchkin <kostap@marvell.com>
Date: Wed, 9 May 2018 18:11:04 +0300
Subject: [PATCH 0631/1051] dts: marvell: enable AP810 system controller for
 A8K Plus family

Change-Id: I09fcdda5c9b970caa46f12c1cce60bf538f7d7a9
Signed-off-by: Konstantin Porotchkin <kostap@marvell.com>
[Kevin: The original patch got from Marvell sdk10.0_19.06]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 arch/arm64/boot/dts/marvell/armada-ap810.dtsi | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/arch/arm64/boot/dts/marvell/armada-ap810.dtsi b/arch/arm64/boot/dts/marvell/armada-ap810.dtsi
index dce37fd7c2d1..9f5033092edd 100644
--- a/arch/arm64/boot/dts/marvell/armada-ap810.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-ap810.dtsi
@@ -91,6 +91,16 @@
 			ranges = <0x0 0x0 AP810_BASE 0x4000000>;
 			interrupt-parent = <&gic>;
 
+			AP810_LABEL(syscon): system-controller@6f4000 {
+				compatible = "syscon", "simple-mfd";
+				reg = <0x6f4000 0x2000>;
+
+				AP810_LABEL(clk): clock {
+					compatible = "marvell,ap810-clock";
+					#clock-cells = <1>;
+				};
+			};
+
 			AP810_LABEL(gicp): gicp@3000040 {
 				compatible = "marvell,ap806-gicp";
 				reg = <0x3000040 0x10>;
-- 
2.17.1

